crates/red_knot_module_resolver/src/path.rs

```diff
@@ -73,6 +73,7 @@ enum ModuleResolutionPathBufInner {
     FirstParty(SystemPathBuf),
     StandardLibrary(FilePath),
     SitePackages(SystemPathBuf),
+    EditableInstall(SystemPathBuf),
```

Comment:
I considered just using the `SitePackages` variant for editable installs, since these search paths do come about as a consequence of `.pth` files in the `site-packages` directory. However, the search paths for editable installs aren't necessarily children of the `site-packages` search path (and indeed most often won't be), and I think it would be too easy to assume that all instances of the `SitePackages` variant have the `site-packages` directory as their root.

Reply:
I don't have a problem with introducing a new enum variant for pth-added paths, though I'm also not totally convinced by it: I don't know when we'd ever handle them differently, and it requires one more chunk of duplicated code wherever we handle this enum. I don't think the confusion you mention should occur: in general search paths should never be children of other search paths (this is a really bad setup that leads to the same module being importable under two different names), so it doesn't seem to me to be a likely confusion to assume this would be the case.

I'm not sure about the name `EditableInstall`. `pth` files are a feature of the Python import system that are _used_ by the editable-install feature of some package managers, but the feature predates that use, and `pth` files have been (and I'm sure still are) used in many other ways as well (e.g. for the zipped-egg installation format of setuptools). So I think it's potentially a little misleading to imply that every path discovered in a `.pth` file is an editable install.

I'd suggest `PthEntry` or something similar instead.

Reply:
> I'm not sure about the name `EditableInstall`. `pth` files are a feature of the Python import system that are _used_ by the editable-install feature of some package managers, but the feature predates that use, and `pth` files have been (and I'm sure still are) used in many other ways as well (e.g. for the zipped-egg installation format of setuptools). So I think it's potentially a little misleading to imply that every path discovered in a `.pth` file is an editable install.

Hrm, true, but the validation we do elsewhere should ensure that we only add paths that point to directories when we iterate through `.pth` files to try to find paths that could potentially be editable installs. While it's true that a directoy path in a `.pth` file could point to something which wasn't intended to be an editable install, the Python runtime will still consider Python modules at that path to be importable, correct (since they'll be added to `sys.path` by `site.py`)? So IIUC a directory path in a `.pth` file is _in effect_ an editable install, even if it... wasn't meant to be.

All of which is to say, from the perspective of the module resolver, this variant should only really be used if we think it's more-likely-than-not that a path represents a directory that has been editably installed, or a path relative to a path representing a directory that has been editably installed. We should have done sufficient validation at this point that it's more than just "an arbitrary line in a `.pth` file".

Reply:
Adding some more docs makes sense but I won't do it _just_ now, since I may still refactor this pretty soon according to the idea Carl suggested on one of my earlier PRs

Reply:
Certainly any path found in a `.pth` file will be added to `sys.path`, and particularly if that path is a directory, it will result in modules there being importable. I'm not sure if there is any other validation you're referring to that we do.

The logical jump you're making that I don't quite follow is "if we add a path to sys.path that makes things importable, that's an 'editable install'." I don't think that follows. It's just importable Python source code. "Editable install" is a very specific user-facing feature that implies a certain use case. For example, there's no reason the path added by the `pth` file line couldn't point to some Python code on a read-only filesystem. Not very editable!